src/OSPSuite.Infrastructure/Services/DataRepositoryTask.cs

```diff
@@ -250,20 +249,32 @@ private static string tableNameFrom(DataColumn baseGrid)
          return dataTable;
       }
 
-      private Cache<DataColumn, IDimension> retrieveDimensionsFor(IReadOnlyList<DataColumn> allColumns, Func<DataColumn, IDimension> dimensionRetriever)
+      private Cache<DataColumn, IDimension> retrieveDimensionsFor(IEnumerable<DataColumn> allColumns, Func<DataColumn, IDimension> dimensionRetriever)
       {
          var cacheDimensionsForColumns = new Cache<DataColumn, IDimension>();
          allColumns.Each(x => cacheDimensionsForColumns.Add(x, dimensionRetriever(x)));
          return cacheDimensionsForColumns;
       }
 
-      private void moveDrugColumnsFirst(List<DataColumn> allColumns)
+      private List<DataColumn> sortColumnsForExport(IEnumerable<DataColumn> allColumns)
       {
-         var allColumnsNotDrug = allColumns.Where(x => !x.QuantityInfo.Type.Is(QuantityType.Drug)).ToList();
-         allColumnsNotDrug.Each(c => allColumns.Remove(c));
 
-         //add them add them again at the end at the beginning
-         allColumnsNotDrug.Each(allColumns.Add);
+         var allDrugColumns = new List<DataColumn>(allColumns);
+
+         var allColumnsNotDrug = allDrugColumns
+            .Where(x => !x.QuantityInfo.Type.Is(QuantityType.Drug))
+            .ToList();
+
+         allColumnsNotDrug.Each(x => allDrugColumns.Remove(x));
+
+         var allAuxiliariesColumns = allColumnsNotDrug
+            .Where(x => x.DataInfo.Origin.IsOneOf(ColumnOrigins.CalculationAuxiliary, ColumnOrigins.ObservationAuxiliary))
+            .ToList();
+
+         allAuxiliariesColumns.Each(x => allColumnsNotDrug.Remove(x));
+
+         //First return the drug columns, then non drug columns that are not auxiliary columns and last auxiliary columns
+         return allDrugColumns.Union(allColumnsNotDrug).Union(allAuxiliariesColumns).ToList();
```

Comment:
There is probably a clever way to do this but I could not think of one

Reply:
What will allColumnsNotDrug contain? Metabolites, enzymes, transporters etc.?
Is this of importance, that allColumnsNotDrug are always exported after allColumnsDrug? (why?)

Reply:
This was a requirement from before. Nothing new here. The goal was to always have the drug name at the beginning because it is what matters when you export a simulation file. 

I don't want to change `THAT` behavior